crates/wasmi/tests/imports/mod.rs

```diff
@@ -0,0 +1,105 @@
+use wasmi::{AsContext, AsContextMut, Engine, Error, Linker, Memory, MemoryType, Module, Store};
+
+#[test]
+fn test_import_one_memory_but_there_are_two_memories() -> Result<(), Error> {
+    let engine = Engine::default();
+    let wat = r#"
+            (module
+                (import "env" "memory" (memory (;0;) 4))
+            )
+        "#;
+    let wasm = wat::parse_str(wat).unwrap();
+    let module = Module::new(&engine, &mut &wasm[..])?;
+
+    let mut store = Store::new(&engine, ());
+
+    let mut linker = <Linker<()>>::new();
+    let mem_type = MemoryType::new(4, None)?;
+    let memory = Memory::new(store.as_context_mut(), mem_type)?;
+    linker.define("env", "memory", memory)?;
+    let instance = linker
+        .instantiate(&mut store, &module)
+        .unwrap()
+        .start(&mut store)?;
+
+    // #0 memory
+    let mem0 = instance.get_memory(store.as_context(), 0).unwrap();
+    // Bug here:
+    // we only import one memory and not define memory inside module.
+    // this should not be defined internally.
+    // #1 memory
+    let mem1 = instance.get_memory(store.as_context(), 1).unwrap();
```

Comment:
This indeed seems to be a problem. If this was a regression test you'd want to assert that `instance.get_memory(store.as_context(), 1)` equals to `None`.